internal/pkg/hub/manifest.go

```diff
@@ -0,0 +1,95 @@
+package hub
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/confluentinc/cli/internal/pkg/ccstructs"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/log"
+)
+
+func (c *Client) GetRemoteManifest(owner, name, version string) (*ccstructs.Manifest, error) {
+	manifestUrl := fmt.Sprintf("%s/api/plugins/%s/%s", c.URL, owner, name)
+	if version != "latest" {
+		manifestUrl = fmt.Sprintf("%s/versions/%s", manifestUrl, version)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, manifestUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Debug {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err != nil {
+			return nil, err
+		}
+		log.CliLogger.Tracef("\n%s\n", string(dump))
+	}
+
+	r, err := c.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Debug {
+		dump, err := httputil.DumpResponse(r, true)
+		if err != nil {
+			return nil, err
+		}
+		log.CliLogger.Tracef("\n%s\n", string(dump))
+	}
+
+	defer r.Body.Close()
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != http.StatusOK {
+		response := make(map[string]interface{})
+		_ = json.Unmarshal(body, &response)
+		if errorMessage, ok := response["message"]; ok {
```

Comment:
Don't we have a general-purpose function that catches HTTP error details?

Reply:
We have some v2 error catchers, but I'm not sure beyond that.